src/app/src/util/hooks.js

```diff
@@ -18,10 +18,35 @@ export default function useUpdateLeafletMapImperatively(
         data,
         shouldPanMapToFacilityDetails,
         isVectorTileMap = false,
+        extent,
+        zoomToSearch,
     } = {},
 ) {
     const mapRef = useRef(null);
 
+    const [
+        currentExtent,
+        setCurrentExtent,
+    ] = useState(extent);
+    useEffect(() => {
+        if (zoomToSearch && extent != null && currentExtent !== extent) {
+            const leafletMap = get(mapRef, 'current.leafletElement', null);
+
+            if (leafletMap) {
+                leafletMap.fitBounds([
+                    [extent[3], extent[2]],
+                    [extent[1], extent[0]],
+                ], { maxZoom: detailsZoomLevel });
```

Comment:
Ran into a literal edge case :-) When searching in Bangladesh there are some markers that are clipped at the edge of the map at my current monitor resolution. Consider adding a buffer here. I tried `[20, 20]`

![2020-02-28 16 16 08](https://user-images.githubusercontent.com/17363/75594420-e89bf680-5a45-11ea-9061-e32ede51a666.gif)

Reply:
This appears to be at least partially related to the dimensions of the viewport when zooming in the map. If I set my window to be tall and fairly narrow, the items at the left/right of the screen get clipped; similarly, if I set my window to be wide and short, the items at the top/bottom of the screen get clipped. Adding the padding prevents the clipping in most circumstances, but when the screen is at an extreme height/width ratio it still occurs. 

Do we want to accept the padding as 'fixes most cases', or should I go back to the drawing board to try to find a different method of handling this?

Reply:
I think that padding is the the right solution for this problem. There are likely some magic numbers that prevent clipping 100% of the time, but I don't think it is worth spending more time trying to optimize it for every case. We can revisit this and adjust it if we get repeated feedback about it being a usability issue.